internal/controllers/machinedeployment/machinedeployment_sync_test.go

```diff
@@ -495,6 +499,224 @@ func TestSyncDeploymentStatus(t *testing.T) {
+		expectedMS.Spec.Replicas = pointer.Int32(2) // 4 (maxsurge+replicas) - 2 (replicas of old ms) = 2
+
+		g := NewWithT(t)
+		actualMS, err := (&Reconciler{}).computeDesiredMachineSet(deployment, nil, []*clusterv1.MachineSet{oldMS}, log)
+		g.Expect(err).To(BeNil())
+		assertMachineSet(g, actualMS, expectedMS)
+	})
+
+	t.Run("should compute the updated MachineSet when no old MachineSets exists", func(t *testing.T) {
+		uniqueID := apirand.String(5)
+		existingMS := skeletonMSBasedOnMD.DeepCopy()
+		existingMSUID := types.UID("abc-123-uid")
+		existingMS.UID = existingMSUID
+		existingMS.Name = deployment.Name + "-" + uniqueID
+		existingMS.Labels = map[string]string{
+			clusterv1.MachineDeploymentUniqueLabel: uniqueID,
+			"ms-label-1":                           "ms-value-1",
+		}
+		existingMS.Annotations = nil
+		existingMS.Spec.Template.Labels = map[string]string{
+			clusterv1.MachineDeploymentUniqueLabel: uniqueID,
+			"ms-label-2":                           "ms-value-2",
+		}
+		existingMS.Spec.Template.Annotations = nil
+		existingMS.Spec.Template.Spec.NodeDrainTimeout = duration5s
+		existingMS.Spec.Template.Spec.NodeDeletionTimeout = duration5s
+		existingMS.Spec.Template.Spec.NodeVolumeDetachTimeout = duration5s
+		existingMS.Spec.DeletePolicy = string(clusterv1.NewestMachineSetDeletePolicy)
+		existingMS.Spec.MinReadySeconds = 0
+
+		expectedMS := skeletonMSBasedOnMD.DeepCopy()
+		expectedMS.UID = existingMSUID
+		expectedMS.Name = deployment.Name + "-" + uniqueID
+		expectedMS.Labels[clusterv1.MachineDeploymentUniqueLabel] = uniqueID
+		expectedMS.Spec.Template.Labels[clusterv1.MachineDeploymentUniqueLabel] = uniqueID
+
+		g := NewWithT(t)
+		actualMS, err := (&Reconciler{}).computeDesiredMachineSet(deployment, existingMS, nil, log)
+		g.Expect(err).To(BeNil())
+		assertMachineSet(g, actualMS, expectedMS)
+	})
+
+	t.Run("should compute the updated MachineSet when old MachineSets exist", func(t *testing.T) {
+		uniqueID := apirand.String(5)
+		existingMS := skeletonMSBasedOnMD.DeepCopy()
+		existingMSUID := types.UID("abc-123-uid")
+		existingMS.UID = existingMSUID
+		existingMS.Name = deployment.Name + "-" + uniqueID
+		existingMS.Labels = map[string]string{
+			clusterv1.MachineDeploymentUniqueLabel: uniqueID,
+			"ms-label-1":                           "ms-value-1",
+		}
+		existingMS.Annotations = nil
+		existingMS.Spec.Template.Labels = map[string]string{
+			clusterv1.MachineDeploymentUniqueLabel: uniqueID,
+			"ms-label-2":                           "ms-value-2",
+		}
+		existingMS.Spec.Template.Annotations = nil
+		existingMS.Spec.Template.Spec.NodeDrainTimeout = duration5s
+		existingMS.Spec.Template.Spec.NodeDeletionTimeout = duration5s
+		existingMS.Spec.Template.Spec.NodeVolumeDetachTimeout = duration5s
+		existingMS.Spec.DeletePolicy = string(clusterv1.NewestMachineSetDeletePolicy)
+		existingMS.Spec.MinReadySeconds = 0
+
+		oldMS := skeletonMSBasedOnMD.DeepCopy()
+		oldMS.Spec.Replicas = pointer.Int32(2)
```

Comment:
Can we add a godoc comment here that even though in this case an oldMS exists and has 2 replicas we don't expect computeDesiredMachineSet to update the replica field? (as it's done elsewhere)